testing/php7-pecl-rrd: new aport

mykeul opened the pull request:
https://pecl.php.net/package/rrd
PHP7 bindings to rrd tool system

Packaging based on Pascal Hofmann's work on PR #1519

Supersedes #1519
Fixes https://gitlab.alpinelinux.org/alpine/aports/issues/6934

maxice8 commented:
the commit title is in the wrong order

```
<package>/<repo>: new aport

$url
$pkgdesc
```

maxice8 commented on testing/php7-rrd/APKBUILD:
```diff
@@ -0,0 +1,38 @@
+pkgdesc="PHP$_phpver bindings to rrd tool system"
+url="https://pecl.php.net/package/rrd"
+arch="all"
+license="BSD"
```
license is not spdx compliant

mykeul replied:
Looks like BSD-2-Clause, with one changed word: "owner" instead of "holder". Is it still BSD-2-Clause ?

maxice8 replied:
i'd say yes

maxice8 commented on testing/php7-rrd/APKBUILD:
```diff
@@ -0,0 +1,38 @@
+depends="php$_phpver-common"
+makedepends="php$_phpver-dev autoconf rrdtool-dev"
+source="$pkgname-$pkgver.tgz::http://pecl.php.net/get/$_pkgreal-$pkgver.tgz"
+options="!check"
```
please add a `# comment` on why it is disabled

maxice8 commented on testing/php7-rrd/APKBUILD:
```diff
@@ -0,0 +1,38 @@
+
+package() {
+	local confdir="$pkgdir/etc/php$_phpver/conf.d"
+	cd "$builddir"
```
cd "$builddir" is not required

maxice8 commented on testing/php7-rrd/APKBUILD:
```diff
@@ -0,0 +1,38 @@
+builddir="$srcdir/$_pkgreal-$pkgver"
+
+build() {
+	cd "$builddir"
```
cd "$builddir" is not required

tcely commented on testing/php7-rrd/APKBUILD:
```diff
@@ -0,0 +1,38 @@
+pkgver=2.0.1
+pkgrel=0
+_phpver=${pkgname#php}
+_phpver=${_phpver%%-*}
```
Where is the value of `_phpver` coming from?

tcely commented on testing/php7-rrd/APKBUILD:
```diff
@@ -0,0 +1,38 @@
+pkgver=2.0.1
+pkgrel=0
+_phpver=${pkgname#php}
+_phpver=${_phpver%%-*}
```
Why not something like this?
```
_phpver=7
_pkgreal=rrd
pkgname="php${_phpver}-${_pkgreal}"
```

mykeul replied:
This was present in the original packaging from Pascal Hofmann. I guess he made this to make the build compatible with any version of php.

andypost commented on testing/php7-rrd/APKBUILD:
```diff
@@ -0,0 +1,40 @@
+# Contributor: Pascal Hofmann <mail@pascalhofmann.de>
+# Contributor: Micka�l Remars <github@remars.com>
+# Maintainer: Pascal Hofmann <mail@pascalhofmann.de>
+pkgname=php7-rrd
```
Name should be `php7-pecl-rrd`

mykeul replied:
Renamed the package. The extension seems still compatible with recent php versions. I just had to install an additional font package to get complete graphs.

andypost commented on testing/php7-pecl-rrd/APKBUILD:
```diff
@@ -0,0 +1,40 @@
+arch="all"
+license="BSD" # Looks like BSD-2-Clause, with just one word difference: "owner" in php-rrd's licence, "holder" in bsd-2-clause
+depends="php$_phpver-common"
+makedepends="php$_phpver-dev autoconf rrdtool-dev"
```
php7-dev already depends on `autoconf` https://pkgs.alpinelinux.org/package/edge/community/x86_64/php7-dev

andypost commented:
I'm not a fan of `$_phpver` replacement but fine
```
checking for re2c... no
configure: WARNING: You will need re2c 0.13.4 or later if you want to regenerate PHP parsers. 
```

better to add it to `makedepends`

Also please squash all commits into single one

tcely commented on testing/php7-pecl-rrd/APKBUILD:
```diff
@@ -0,0 +1,38 @@
+makedepends="php7-dev re2c rrdtool-dev"
+checkdepends="rrdtool"
+source="$pkgname-$pkgver.tgz::http://pecl.php.net/get/$_pkgreal-$pkgver.tgz"
+options="check"
```
```suggestion
```

mykeul commented:
renamed to php7-pecl-rrd
removed not needed "cd" lines
updated license
added check() and removed "options="
removed $_phpver
removed autoconf from makedepends and added re2c
squashed all commits in a single one (and adjusted commit message)

tcely commented:
> the commit title is in the wrong order
> 
> ```
> <package>/<repo>: new aport
> 
> $url
> $pkgdesc
> ```

Please follow the prescribed format and put a blank line after the description before adding any additions, such as which pull request this is based on.

tcely commented on testing/php7-pecl-rrd/APKBUILD:
```diff
@@ -0,0 +1,37 @@
+# Contributor: Micka�l Remars <github@remars.com>
+# Maintainer: Pascal Hofmann <mail@pascalhofmann.de>
+pkgname=php7-pecl-rrd
+_pkgreal=rrd
```
I would put `_pkgreal` above `pkgname` but that's just my preference.

maxice8 commented:
Fixes https://gitlab.alpinelinux.org/alpine/aports/issues/6934

andypost commented:
Only nitpick is missing eol at the end of file

tcely commented:
Adjusted the commit message, encoding and added missing end of line.
